src/main/java/eu/dissco/core/translator/service/BioCaseService.java

```diff
@@ -230,6 +232,18 @@ private void processUnit(DataSet dataset, Unit unit, AtomicInteger processedReco
     }
   }
 
+  private List<DigitalMediaEvent> makeDigitalMediaUnique(List<DigitalMediaEvent> digitalMedia) {
+    return digitalMedia.stream()
+        .collect(Collectors.toMap(
+            mediaEvent -> mediaEvent.digitalMedia().attributes().getAcAccessURI(),
+            mediaEvent -> mediaEvent,
+            (existing, replacement) -> replacement,
+            LinkedHashMap::new))
+        .values()
+        .stream()
+        .toList();
```

Comment:
returning a set would make it clear that this collection only has unique items

Reply:
Hmm I rather keep this as a List. Making it a Set doesn't force it to have unique items as the uniqueness should be based on the accessURI and not the whole object. We could get two different MediaEvents with the same accessURI something that the Set wouldn't catch. Also changing this to a Set forces us to make a lot of excalating changes as the DigitalSpecimenEvent will then contain a Set which means changes to the processors as well. So it wouldn't really solve any issue and the indication that it is unique because it is a Set is invalid.

Reply:
well put, sounds fair.